src/sentry/auth/staff.py

```diff
@@ -0,0 +1,319 @@
+from __future__ import annotations
+
+import ipaddress
+import logging
+from datetime import datetime, timedelta
+
+from django.conf import settings
+from django.core.signing import BadSignature
+from django.utils import timezone
+from django.utils.crypto import constant_time_compare, get_random_string
+from rest_framework.request import Request
+
+from sentry.auth.elevated_mode import ElevatedMode
+from sentry.auth.system import is_system_auth
+from sentry.utils.auth import has_completed_sso
+
+logger = logging.getLogger("sentry.staff")
+
+SESSION_KEY = "_stf"
```

Comment:
Is there a character limit? Can we name this `_staff`?

Reply:
There's no character limit, I chose `_stf` because superuser uses `su` for the cookie name. Do you want to switch the session and cookie over to `staff` instead of `stf`?